File: flow/backend/internal/core/domain/entity/node.go
package entity

import "github.com/kyh0703/flow/internal/pkg/constant"

type Group struct {
	ID     string `json:"id"`
	Extent bool   `json:"extent"`
	Expand string `json:"expand"`
}
type Position struct {
	X int32 `json:"x"`
	Y int32 `json:"y"`
}

type Style struct {
	BgColor     string `json:"bgColor"`
	Color       string `json:"color"`
	BorderStyle string `json:"borderStyle"`
	BorderColor string `json:"borderColor"`
	Hidden      bool   `json:"hidden"`
}

type Node struct {
	ID        string            `json:"id"`
	SubFlowID int64             `json:"subFlowId"`
	Type      constant.NodeType `json:"type"`
	Width     int32             `json:"width"`
	Height    int32             `json:"height"`
	Label     string            `json:"label"`
	Group     Group             `json:"group"`
	Position  Position          `json:"position"`
	Style     Style             `json:"style"`
	Desc      string            `json:"desc"`
}

File: flow/frontend/src/models/node.ts
import type { CustomNodeType } from '@xyflow/react'

type PositionXY = {
  x: number
  y: number
}

type Style = {
  bgColor: string
  color: string
  borderStyle: string
  borderColor: string
  width: number
  height: number
  hidden: boolean
}

type Parent = {
  width: number
  height: number
  collapsed: boolean
}

export interface Node {
  id: string
  flowId: number
  type: CustomNodeType
  label: string
  position: PositionXY
  style: Style
  parent: Parent
  description: string
}
